File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/EnemyTargetConstraint.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints;

import com.anathema_roguelike.entities.characters.foes.ai.Faction;

public class EnemyTargetConstraint implements TargetConstraint<Character, Character> {
	
	@Override
	public boolean apply(Character target, Character character) {
		return !Faction.friendly(target, character);
	}

}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/EnemyTargetConstraint.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.constraints

import com.anathema_roguelike.entities.characters.foes.ai.Faction
import com.anathema_roguelike.entities.characters.Character

class EnemyTargetConstraint extends TargetConstraint[Character, Character] {
  override def apply(target: Character, character: Character): Boolean = !Faction.friendly(target, character)
}